stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -263,23 +316,32 @@ RemoteRef<FieldDescriptor> TypeRefBuilder::getFieldTypeInfo(const TypeRef *TR) {
   if (Found != FieldTypeInfoCache.end())
     return Found->second;
 
+  if (auto FD = getFieldDescriptorFromProvider(*MangledName, ExternalTypeInfo))
+    return *FD;
+
   // Heuristic: find the outermost Module node available, and try to parse the
   // ReflectionInfos with a matching name first.
   auto ModuleName = FindOutermostModuleName(Node);
   // If we couldn't find a module name or the type is imported (__C module) we
   // don't any useful information on which image to look for the type.
   if (ModuleName && ModuleName != llvm::StringRef("__C"))
     for (size_t i = 0; i < ReflectionInfos.size(); ++i)
-      if (llvm::is_contained(ReflectionInfos[i].PotentialModuleNames,
-                             ModuleName))
-        if (auto FD = findFieldDescriptorAtIndex(i, *MangledName))
-          return *FD;
+      if ((!ExternalTypeInfo ||
+           ExternalTypeInfo->shouldParseReflectionInfoWithId(i)))
+        if (llvm::is_contained(ReflectionInfos[i].PotentialModuleNames,
```

Comment:
Not sure if this is much better, bug have you considered a SmallSet for PotentialModuleNames (in the small case it's also a linear list).

Reply:
This unfortunately can't be a SmallSet because the stdlib doesn't ship with it. I'm not even sure a set would be that much faster since there are so few elements.